### PR TITLE
docs(13881): docs for logical layer 

### PR DESCRIPTION
**Pull request by Akiyamka**

### CODE REVIEW

File: src/core/logical_layers/types/logicalLayer.ts [hunk 1]
```diff
@@ -7,7 +7,9 @@ import type { LayerContextMenu } from './contextMenu';
 
 export type LogicalLayerState = Readonly<{
   id: string;
+  /* This state means - we need to show this layer */
   isEnabled: boolean;
+  /* This state means - we show this layer on map */
```
Comment (IlyaIzr) [inline src/core/logical_layers/types/logicalLayer.ts]:
```suggestion
  /* isEnabled tells if we need to show this layer */
  isEnabled: boolean;
  /* isMounted means the layer is already shown on the map */
```

File: src/core/logical_layers/rfc.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Layer types
+
+Before answering the question what are logical layers, we should first consider what kind of layers we have in the context of a map.
+
+## Map Style Layers
+
+The [Style Specification](https://maplibre.org/maplibre-gl-js-docs/style-spec/) defines a map layer as a way of representing data.
+Such a layer is part of the map style and can be of type "fill", "line", "symbol", "circle" and so on.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
Such layer is part of the map style and can be of type "fill", "line", "symbol", "circle" and so on.
```

File: src/core/logical_layers/rfc.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+In the context of our application, we refer to such layers as map style layers, or just a style layer
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
In the context of our application, we refer to such layers as map style layers, or just a `style layer`
```

File: src/core/logical_layers/rfc.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+## Sources of layers
+
+If we divide layers by data delivery format, we can identify "vector", "raster" and "geojson" layers.
+However, this is a simplification - despite the fact that it directly affects the way the layer style is described, this division still refers to _layer sources_.
+At the same time, the vector data source type also has layers, which we call source-layer
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]: I don't get the message of this paragraph. Can you rephrase that as simple as possible?

File: src/core/logical_layers/rfc.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Logical layers
+
+And, at the top of the list are the logical layers. A logical layer is a concatenation of several data and style layers linked together by a common theme.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
Finally - the `logical layer` is a concatenation of several data and style layers linked together by a common theme.
```

File: src/core/logical_layers/rfc.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+This is how these layers look like a tree:
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
Here's a tree structure example of a logical layer:
```

File: src/core/logical_layers/rfc.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+- logical layer
+  - style layers
+  - layer source
+    - source layer
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md:29]:
```suggestion
- logical layer
  - style layer #1
  - layer #1 sources 
    - source layer
  - style layer #2
  - layer #2 sources
    - source layer (a)
    - source layer (b)
```
couple of suggestions
also, is that logic applicable?

File: src/core/logical_layers/rfc.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+For the user of an application, the logical layer is the layer with which he interacts.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
For the user of an application, the logical layer is the layer he interacts with.
```

File: src/core/logical_layers/rfc.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+For example, the logical layer "active contributors" includes:
+
+- a vector data source with multiple data layers (hexagon geometry h3, and coordinates with nicknames)
+- three style layers - text style, hexagon display style, glow style under the text.
+
+# Structure of a logical layer
+
+Each logical layer has additional data not related to the map, but necessary for other parts of the interface interacting with it.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md:48]:
```suggestion
Each logical layer has additional data not related to the map, necessary for other parts of the interface interacting with it.
```

File: src/core/logical_layers/rfc.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+The logical layer interface can be found in the file [types/logicalLayers.ts](types/logicalLayer.ts).
+You can notice that we can divide them conditionally into two halves: data and states.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]: "them" who?
```suggestion
You can notice it's properties can be seen as two semantic parts: data and states.
```

File: src/core/logical_layers/rfc.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+The internal logic of the logical layer is described in [utils/logicalLayerFabric.ts](utils/logicalLayerFabric.ts)
+
+## Layers registry
+
+Depending on the application's task, you need to track changes in _all_ states of _one_ layer, or _specified_ state of _all_ layers. For example, an item in the Layer List tracks all changes in its layer, while a legend panel tracks changes in the legends of all layers.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
Depending on the application's task, you need to track changes in _all_ states of _one_ layer, or _specific_ state of _all_ layers. For example, an item in the Layer List tracks all changes in its layer, while a legend panel tracks changes in the legends of all layers.
```

File: src/core/logical_layers/rfc.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+For this reason, states are stored in separate lists (layersMeta, layersLegends, layersMenus, enabledLayers, mountedLayers, etc.) from which each layer selects the state relevant to it.
+
+Separately stands layersRegistry - a list that contains all the logical layers, knows how to create and delete them.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
`layersRegistry` is a special list, that contains all the logical layers and knows how to create and delete them.
```

File: src/core/logical_layers/rfc.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+It is important to note that you do not need to add a logical layer to it, this should be done only if you want to allow the user to manage this layer in a standard way through the panel with the layers
+
+### isEnabled and isMounted
+
+The isEnabled and isMounted states deserves special mention. The layer gets `isEnabled` state if we want to show the layer to the user but for some reason cannot do it right now. Its main purpose is to _remember the choices made by the user_.
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
The `isEnabled` and `isMounted` states deserves special mention. The layer gets `isEnabled` state if we want to show the layer to the user. Its main purpose is to _remember the choices made by the user_.
```
isEnabled stays on after layer being mounted, and this part implied that it turns off once layer was mounted

File: src/core/logical_layers/rfc.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+The moment we managed to get all the necessary data and layer can be showed on the map - layer calls the hook willMount from its renderer and becomes isMounted
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
The moment we managed to get all the necessary data and layer can be showed on the map - layer calls the hook `willMount` from its renderer and becomes `isMounted`
```

File: src/core/logical_layers/rfc.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+For example - if the user enable logical layer "X", then limited the search area of layers to a certain geometry,
+which it does not fall into this area - the layer will disappear from the list of available layers, and it will be removed from the map.
+However, as soon as the conditions change and we have enough data to display the layer - the layer will appear in the list and map again, without secondary "enable" action from the user
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
For example - if user enables logical layer "X", then limits the search area out of layer's scope - the layer will disappear from the list of available layers, and it will be removed from the map.
However, as soon as the conditions change and we have enough data to display the layer - the layer will appear in the list and on the map again, without secondary "enable" action from the user
```

File: src/core/logical_layers/rfc.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+This mechanism is described inside logical layer atom [utils/logicalLayerFabric.ts](utils/logicalLayerFabric.ts#L189)
+
+### Style and Legend
+
+The layer has such a property as a legend. The legend gives the user a hint about how the data is labeled on the map. However - we use _the same data_ to generate style layers, which allows us to have one source of truth
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
The layer has `legend` property. The legend gives user a hint about how the data is labeled on the map. However - we also use _the same data_ to generate style layers, which allows us to have one source of truth
```

File: src/core/logical_layers/rfc.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+## Layers hierarchy
+
+layersHierarchy contains a list of layers containing only those layer properties that are needed to build the tree and define the position of the layer in the hierarchy
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
layersHierarchy contains a list of layers having only properties that are needed to build the tree and define the position of the layer in the hierarchy
```

File: src/core/logical_layers/rfc.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+
+## Layers renderers
+
+Every logical layer must be rendered on the map sooner or later.
+Layer renderer is a class that describes how to do this. By default [GenericRenderer] (renderers/GenericRenderer.ts) is used, but in case it is not suitable - another renderer can be defined for the layer, which implements [Renderer](renderers/DefaultRenderer.ts) interface, the essence of which is to describe the map reaction to various events of the logical layer life cycle
```
Comment (IlyaIzr) [inline src/core/logical_layers/rfc.md]:
```suggestion
Layer renderer is a class that describes how to do this. By default [GenericRenderer](renderers/GenericRenderer.ts) is used, but in case it is not suitable - another renderer can be defined for the layer, which implements [Renderer](renderers/DefaultRenderer.ts) interface, the essence of which is to describe the map reaction to various events of the logical layer life cycle
```

File: src/core/logical_layers/rfc.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
+# Layer types
+
+Before answering the question what are logical layers, we should first consider what kind of layers we have in the context of a map.
```
Comment (propakov) [inline src/core/logical_layers/rfc.md]:
```suggestion
Before answering the question what logical layers are, we should at first consider what kind of layers we have in the context of a map.
```

File: src/core/logical_layers/rfc.md (hunk 3 above)
Comment (propakov) [reply src/core/logical_layers/rfc.md]: 'map style layers' also surround with ``  ?

File: src/core/logical_layers/rfc.md (hunk 7 above)
Comment (Akiyamka) [reply src/core/logical_layers/rfc.md:29]: I just want to show hot they related to each other, this is most important part of logical layer purpose. Real world example listed right after this tree

File: src/core/logical_layers/rfc.md (hunk 6 above)
Comment (Akiyamka) [reply src/core/logical_layers/rfc.md]: How about "This is how these layers nested" ?

File: src/core/logical_layers/rfc.md (hunk 9 above)
Comment (Akiyamka) [reply src/core/logical_layers/rfc.md:48]: What wrong with `but` ?

File: src/core/logical_layers/rfc.md (hunk 4 above)
Comment (Akiyamka) [reply src/core/logical_layers/rfc.md]: was rewritten

File: src/core/logical_layers/rfc.md (hunk 6 above)
Comment (IlyaIzr) [reply src/core/logical_layers/rfc.md]: Yes!

File: src/core/logical_layers/rfc.md (hunk 9 above)
Comment (IlyaIzr) [reply src/core/logical_layers/rfc.md:48]: my bad, didn't see "not"

File: src/core/logical_layers/rfc.md (hunk 7 above)
Comment (IlyaIzr) [reply src/core/logical_layers/rfc.md:29]: but not in a tree structure

Comment (IlyaIzr) [reply src/core/logical_layers/rfc.md:29]: your choise

Comment (Akiyamka) [reply src/core/logical_layers/rfc.md:29]: real world tree added

**Review by IlyaIzr — APPROVED** [review]
